extensions/tracing/src/main/java/io/opentelemetry/extension/tracing/ExtendedTracer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.tracing;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import java.util.concurrent.Callable;
+
+/** Provides easy mechanisms for wrapping standard Java constructs with an OpenTelemetry Span. */
+@SuppressWarnings("InconsistentOverloads")
+public class ExtendedTracer {
+
+  private final Tracer tracer;
+
+  /** Create a new {@link ExtendedTracer} that wraps the provided Tracer. */
+  public ExtendedTracer(Tracer tracer) {
+    this.tracer = tracer;
+  }
+
+  /** Wrap the provided {@link Runnable} with a {@link Span} with the provided name. */
+  public Runnable wrap(String spanName, Runnable runnable) {
+    return wrap(tracer, spanName, runnable);
+  }
+
+  /**
+   * Wrap the provided {@link Runnable} with a {@link Span} with the provided name. Uses the
+   * provided {@link Tracer} for the functionality.
+   */
+  public static Runnable wrap(Tracer tracer, String spanName, Runnable runnable) {
```

Comment:
We can remove these statics by implementing `Tracer`, it's free to just use `ExtendedTracer` instead of `Tracer`

Reply:
done